packages/views/src/use-view.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import { dequal } from 'dequal';
+
+/**
+ * Internal dependencies
+ */
+import { generatePreferenceKey } from './preference-keys';
+import type { ViewConfig } from './types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useMemo } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import type { View } from '@wordpress/dataviews';
+// @ts-ignore - Preferences package is not typed
+import { store as preferencesStore } from '@wordpress/preferences';
+
+interface UseViewReturn {
+	view: View;
+	isModified: boolean;
+	updateView: ( newView: View ) => void;
+	resetToDefault: () => void;
+}
+
+function omit< T extends object, K extends keyof T >(
+	obj: T,
+	keys: K[]
+): Omit< T, K > {
+	const result = { ...obj };
+	for ( const key of keys ) {
+		delete result[ key ];
+	}
+	return result;
+}
+
+/**
+ * Hook for managing DataViews view state with local persistence.
+ *
+ * @param config                     Configuration object for loading the view.
+ * @param config.kind                Entity kind (e.g., 'postType', 'taxonomy', 'root').
+ * @param config.name                Specific entity name.
+ * @param config.slug                View identifier.
+ * @param config.defaultView         Default view configuration.
+ * @param config.queryParams         Object with `page` and/or `search` from URL.
+ * @param config.onChangeQueryParams Optional callback to update URL parameters.
+ *
+ * @return Object with current view, modification state, and update functions.
+ */
+export function useView( config: ViewConfig ): UseViewReturn {
+	const { kind, name, slug, defaultView, queryParams, onChangeQueryParams } =
+		config;
+
+	const preferenceKey = generatePreferenceKey( kind, name, slug );
+	const persistedView: View | undefined = useSelect(
+		( select ) => {
+			return select( preferencesStore ).get(
+				'core/views',
+				preferenceKey
+			) as View | undefined;
+		},
+		[ preferenceKey ]
+	);
+	const { set } = useDispatch( preferencesStore );
+
+	const baseView: View = persistedView ?? defaultView;
+	const page = queryParams?.page ?? baseView.page ?? 1;
```

Comment:
For now I guess it is ok but in the future we may need to have an argument specifying which options are url based ['page', 'search'] and their defaults.

Reply:
No, I think it's better to be opinionated here.